### PR TITLE
Refactor key package for app CRs

### PR DESCRIPTION
**Pull request by rossf7**

### CODE REVIEW

File: service/controller/app/resource/appcatalog/create.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ import (
 
 // EnsureCreated gets the appCatalog CR specified in the provided app CR.
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
-	cr, err := key.ToCustomResource(obj)
+	cr, err := key.ToApp(obj)
```
Comment (rossf7) [inline service/controller/app/resource/appcatalog/create.go:13]: Package got flattened to support app and appcatalog CRs. `ToAppCatalog` already exists so aligning the naming.

File: service/controller/key/app.go [hunk 2]
```diff
@@ -41,14 +41,10 @@ func AppCatalogSecretNamespace(customResource v1alpha1.AppCatalog) string {
 	return customResource.Spec.Config.Secret.Namespace
 }
 
-// AppConfigMapName returns the name of the configmap that stores app level
-// config for the provided app CR.
```
Comment (rossf7) [inline service/controller/key/app.go:45]: Some of the key funcs have godoc and others don't. I'd like this to be consistent.

The names are clear so I think we should just drop the godoc.

File: service/controller/key/app.go [hunk 3]
```diff
@@ -73,6 +65,26 @@ func AppStatus(customResource v1alpha1.App) v1alpha1.AppStatus {
...
+	}
+	s := fmt.Sprintf("%s=%s", label.AppOperatorVersion, version)
+
+	selector, err := labels.Parse(s)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse selector %#q with error %#q", s, err))
+	}
+
+	return selector
+}
```
Comment (rossf7) [inline service/controller/key/app.go:86]: Last change just fixes the ordering.

**Review by rossf7 — COMMENTED** [review]
Tested on `ghost`.

**Review by tomahawk28 — APPROVED** [review]

**Pull request merged**